birdvoxpaint/birdvoxpaint.py

```diff
@@ -1,76 +1,110 @@
+from functools import partial
+
 import librosa
-from librosa.util.exceptions.ParameterError
+import librosa.display
 import numpy as np
 
+import tqdm
+import joblib
+
+from . import util
+from .indices import *
+from .display import *
+
+MAX_JOBS = 40
+
+def transform(filename=None, y=None, sr=None,
+              frame_length=2048, hop_length=512,
+
+              n_mels=256, fmin=None, fmax=None,
+              time_constant=0.4, use_pcen=True, use_mels=True,
+
+              indices=[average_energy],
+              segment_duration=10,
+              verbose=True, n_jobs=None, return_plot=False):
+    '''Convert audio into a false color spectrogram.
+
+    Arguments:
+        
+    '''
+
+    if not n_jobs or n_jobs < 0: # automatically use available cpus, limited by MAX_JOBS
+        n_jobs = min(joblib.cpu_count(), MAX_JOBS)
+
+    # load audio frames - generates one block at a time
+    y_blocks, sr = util.block_stream(
+        filename, y, sr,
+        frame_length=frame_length, hop_length=hop_length,
+        segment_duration=segment_duration, n_blocks=n_jobs) # yields: batch*time*n_fft
+
+    y_blocks = tqdm.tqdm(y_blocks, disable=not verbose)
```

Comment:
maybe don't give it the same name. in my book, variables ending in the plural are array-like. But now `y_blocks` is no longer indexable after you apply `tqdm.tqdm` to it.
I would prefer `y_blocks_tqdm` if you must

Reply:
Well it wouldn't be indexable in the first place if you passed a filename? So maybe there's a better way to name it.